### PR TITLE
Updates documentation

### LINKED ISSUE #57: Docs: Documentation for the "phel" configuration section needs to be written
We need to document this: https://github.com/jenshaase/phel-lang/issues/52

> In order to archive this, each library needs to provide a configuration with a namespace. I will suggest that we will put this configuration in the `composer.json` file under `extra.phel`.

```
{
    "extra": {
        "phel": {
            "loader": {
                "phel\\": "src/phel/"
            },
            "loader-dev": {
                "phel\\": "tests/phel/"
            }
        }
    }
}
```

> I also added it as dependency to Phel and tested it with a sample project. Make sure you add `"minimum-stability": "dev"` to your tests-project's composer file. Otherwise you will get an error.

Link to composer plugin: https://github.com/jenshaase/phel-composer-plugin

### LINKED ISSUE #58: Docs: The Getting Started Guide must be changed
We need to document this: https://github.com/jenshaase/phel-lang/issues/52

Update:
- new index.php
- composer phel configuration
- mention running from command line

### LINKED ISSUE #61: Docs: Write documentation for Phel run command
We need to document this: https://github.com/jenshaase/phel-lang/issues/45

```
vendor/bin/phel run path/to/file.phel
# or
vendor/bin/phel run my-namespace\\module
# or
vendor/bin/phel run "my-namespace\module"
```

More info on the run command here: https://github.com/jenshaase/phel-lang/issues/60#issuecomment-645814670

### PR DESCRIPTION
**Pull request by mabasic**
# Description

- Adds composer configuration to documentation. Fixes #57. 
- Updates getting started documentation. Fixes #58. 
- Writes documentation for `phel run`command. Fixes #61.

### CODE REVIEW

File: doc/content/documentation/getting-started.md [hunk 1]
```diff
@@ -58,6 +59,36 @@ Next we can require Phel as dependency of our project.
 composer require phel/phel:dev-master
 ```
 
+After adding Phel as a dependency we can now write some code. First we create new directory `src` and add two files in this directory.
```
Comment (jenshaase) [inline doc/content/documentation/getting-started.md:62]: we, we, we :)

File: doc/content/documentation/getting-started.md [hunk 2]
```diff
@@ -58,6 +59,36 @@ Next we can require Phel as dependency of our project.
...
+The file `boot.phel` contains our actual code of the project. It just defines the namespace and prints "Hello, World!".
+
+```phel
+# in src/boot.phel
+(ns hello-world\boot)
+
+(println "Hello, World!")
+```
+
+Before we can run the code, we need to add Phel specific configuration to `composer.json` file:
```
Comment (jenshaase) [inline doc/content/documentation/getting-started.md:77]: we, we, we :)

**Comment by mabasic** [comment]
@jenshaase I've implemented the changes that you suggested. Let me know what else needs changing. It is better to have more eyes on this text since it will be the official documentation.

File: doc/content/documentation/configuration.md [hunk 3]
```diff
@@ -0,0 +1,91 @@
++++
+title = "Configuration"
+weight = 15
++++
+
+Phel comes with some configuration options. They are stored in the `composer.json` file under `extra`.
```
Comment (jenshaase) [inline doc/content/documentation/configuration.md:6]: under `extra` => in the `extra` section

File: doc/content/documentation/configuration.md [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+## Options
+
+This chapter contains all configuration options explained in detail.
+
+
+### `loader`
+
+Autoload mapping for a Phel autoloader.
+
+Under the `loader` key you define a mapping from namespaces to paths, relative to the package root. When autoloading a module like `hello-world\boot` a namespace prefix `hello-world` pointing to a directory `src/` means that the autoloader will look for a file named `src/boot.phel` and include it if present.
```
Comment (jenshaase) [inline doc/content/documentation/configuration.md:40]: The `loader` configuration defines a mapping from namespaces to paths. The paths are relative to the package root.

File: doc/content/documentation/configuration.md [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+
+Namespace prefixes must end in `\\` to avoid conflicts between similar prefixes. For example `hello` would match modules in the `hello-world` namespace so the trailing backslashes solve the problem: `hello\\` and `hello-world\\` are distinct.
```
Comment (jenshaase) [inline doc/content/documentation/configuration.md:42]: Namespace prefixes must end with a backslash (`\\`) ...

File: doc/content/documentation/configuration.md [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+
+The `loader` references are all added, during install/update, to the Phel runtime which may be found in the generated file `vendor/PhelRuntime.php`.
```
Comment (jenshaase) [inline doc/content/documentation/configuration.md:44]: may be found => can be found

File: doc/content/documentation/configuration.md [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+
+
+### `loader-dev`
+
+This section allows to define autoload rules for development purposes.
+
+Modules needed to run the test suite should not be included in the main autoload rules to avoid polluting the autoloader in production and when other people use your package as a dependency.
```
Comment (jenshaase) [inline doc/content/documentation/configuration.md:51]: other people use your package => other people the package

File: doc/content/documentation/configuration.md [hunk 8]
```diff
@@ -0,0 +1,91 @@
...
+
+Therefore, it is a good idea to rely on a dedicated path for your unit tests and to add it within the `loader-dev` section.
```
Comment (jenshaase) [inline doc/content/documentation/configuration.md:53]: No new paragraph here. "Therefore" should be behind the last sentence.

File: doc/content/documentation/configuration.md [hunk 9]
```diff
@@ -0,0 +1,91 @@
...
+Autoload mapping for a Phel autoloader.
+
+Under the `loader` key you define a mapping from namespaces to paths, relative to the package root. When autoloading a module like `hello-world\boot` a namespace prefix `hello-world` pointing to a directory `src/` means that the autoloader will look for a file named `src/boot.phel` and include it if present.
+
+Namespace prefixes must end in `\\` to avoid conflicts between similar prefixes. For example `hello` would match modules in the `hello-world` namespace so the trailing backslashes solve the problem: `hello\\` and `hello-world\\` are distinct.
+
+The `loader` references are all added, during install/update, to the Phel runtime which may be found in the generated file `vendor/PhelRuntime.php`.
+
+
+### `loader-dev`
```
Comment (jenshaase) [inline doc/content/documentation/configuration.md:47]: The explanation is this section is correct. But you need to add that the configuration is identical to `loader`.

File: doc/content/documentation/configuration.md [hunk 10]
```diff
@@ -0,0 +1,91 @@
...
+This section allows to define autoload rules for development purposes.
+
+Modules needed to run the test suite should not be included in the main autoload rules to avoid polluting the autoloader in production and when other people use your package as a dependency.
+
+Therefore, it is a good idea to rely on a dedicated path for your unit tests and to add it within the `loader-dev` section.
+
+
+## Phel Composer Plugin
+
+Phel runtime is configured automatically by the plugin. On package install/update it generates a file `vendor/PhelRuntime.php` which initializes the Phel runtime and adds package namespaces defined in `loader` and `loader-dev`.
```
Comment (jenshaase) [inline doc/content/documentation/configuration.md:58]: On package install/update... => Whenever a package is updated or installed a file is generated in `vendor/PhelRuntime.php`. This file initializes the Phel runtime according to the defined `loader` and `loader-dev` configuration options.

File: doc/content/documentation/configuration.md [hunk 11]
```diff
@@ -0,0 +1,91 @@
...
+
+Example on how to load the generated runtime.
```
Comment (jenshaase) [inline doc/content/documentation/configuration.md:60]: Better: The generated runtime can be loaded like this.

File: doc/content/documentation/configuration.md [hunk 12]
```diff
@@ -0,0 +1,91 @@
...
+```php
+// src/index.php
+<?php
+
+$rt = require __DIR__ .'/../vendor/PhelRuntime.php';
+
+$rt->loadNs('hello-world\boot');
+```
+
+View source code of the [Composer plugin for the Phel language](https://github.com/jenshaase/phel-composer-plugin).
```
Comment (jenshaase) [inline doc/content/documentation/configuration.md:71]: The source of [Phel's composer plugin](https://github.com/jenshaase/phel-composer-plugin) can be found in a separated repository.

File: doc/content/documentation/configuration.md (hunk 9 above)
Comment (mabasic) [reply doc/content/documentation/configuration.md:47]: I don't know what you mean. Can you explain a bit?

Comment (jenshaase) [reply doc/content/documentation/configuration.md:47]: Something like:

> The `loader-dev` configuration section is equivalent to the `loader` configuration section. Namespaces and paths are defined in the same way.

Comment (mabasic) [reply doc/content/documentation/configuration.md:47]: Going to use exactly that :)

**Comment by jenshaase** [comment]
Deploy the website. Its live now.

**Pull request merged**